test/e2e/framework/elasticsearch_utils.go

```diff
@@ -0,0 +1,189 @@
+)
+
+const (
+	// Poll interval for ElasticsearchCluster tests
+	ElasticsearchClusterPoll = 10 * time.Second
+	// Timeout interval for ElasticsearchCluster operations
+	ElasticsearchClusterTimeout = 10 * time.Minute
+)
+
+// DeleteAllElasticsearchClusters deletes all ElasticsearchCluster API Objects in Namespace ns.
+func DeleteAllElasticsearchClusters(c clientset.Interface, ns string) {
+	esList, err := c.NavigatorV1alpha1().ElasticsearchClusters(ns).List(metav1.ListOptions{LabelSelector: labels.Everything().String()})
+	ExpectNoError(err)
+
+	errList := []string{}
+	for i := range esList.Items {
+		es := &esList.Items[i]
+		Logf("Deleting elasticsearchcluster %v", es.Name)
+		// Use OrphanDependents=false so it's deleted synchronously.
+		// We already made sure the Pods are gone inside Scale().
+		if err := c.NavigatorV1alpha1().ElasticsearchClusters(es.Namespace).Delete(es.Name, &metav1.DeleteOptions{OrphanDependents: new(bool)}); err != nil {
+			errList = append(errList, fmt.Sprintf("%v", err))
+		}
+	}
+
+	if len(errList) != 0 {
+		ExpectNoError(fmt.Errorf("%v", strings.Join(errList, "\n")))
+	}
+}
+
+// ElasticsearchTester is a struct that contains utility methods for testing
+// ElasticsearchCluster related functionality. It uses a clientset.Interface to
+// communicate with the API server.
+type ElasticsearchTester struct {
+	kubeClient kubernetes.Interface
+	navClient  clientset.Interface
+}
+
+func NewElasticsearchTester(kubeClient kubernetes.Interface, navClient clientset.Interface) *ElasticsearchTester {
+	return &ElasticsearchTester{kubeClient, navClient}
+}
+
+func (e *ElasticsearchTester) CreateClusterAndWaitForReady(es *v1alpha1.ElasticsearchCluster) *v1alpha1.ElasticsearchCluster {
+	var err error
+	By("Creating elasticsearchCluster " + es.Name + " in namespace " + es.Namespace)
+	es, err = e.navClient.NavigatorV1alpha1().ElasticsearchClusters(es.Namespace).Create(es)
+	Expect(err).NotTo(HaveOccurred())
+	By("Waiting for CreateNodePool event")
+	WaitForElasticsearchClusterEvent(e.kubeClient, es.Name, es.Namespace, "CreateNodePool")
+	By("Waiting for cluster pods to be ready")
+	e.WaitForAllReady(es)
+	return es
+}
+
+func (e *ElasticsearchTester) ScaleNodePool(es *v1alpha1.ElasticsearchCluster, pool string, replicas int32) {
+	var err error
+	By("Retrieving an up to date copy of the cluster resource")
+	es, err = e.navClient.NavigatorV1alpha1().ElasticsearchClusters(es.Namespace).Get(es.Name, metav1.GetOptions{})
+	Expect(err).NotTo(HaveOccurred())
+	found := false
+	for i, np := range es.Spec.NodePools {
+		if np.Name == pool {
+			np.Replicas = replicas
+			es.Spec.NodePools[i] = np
+			found = true
+			break
+		}
+	}
+	if !found {
+		Failf("Node pool %q not found in ElasticsearchCluster %q", pool, es.Name)
+	}
+	By("Scaling node pool " + es.Name + "/" + pool + " in namespace " + es.Namespace)
+	es, err = e.navClient.NavigatorV1alpha1().ElasticsearchClusters(es.Namespace).Update(es)
```

Comment:
Is this susceptible to conflict errors? If Navigator has since updated the status of the ES cluster resource?
Does this need `RetryOnConflict` handling?